tools/clang/include/clang/Basic/Attr.td

```diff
@@ -984,6 +984,40 @@ def HLSLNodeTrackRWInputSharing : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+def HLSLNodeRecord : InheritableAttr {
```

Comment:
Not all these are technically "Record" objects, so the naming seems slightly misleading.  Maybe "HLSLNodeObject" might be more appropriate.  Also, wouldn't this be something we quickly want to expand to use on all HLSL object types generally?  Could we do that by inheriting this from a more general "HLSLObject" attribute in the future perhaps?

Reply:
Yes, we could have an inheritance structure of these attributes that would allow for more general application.

In terms of expanding this to other HLSL objects, I think that depends a lot on resourcing. There are clearly bugs caused by our string matching (#5738 is a recent example), but cost-benefit wise I'm not sure where this will fit on our schedule.